### PR TITLE
Enable j.u.Objects#requireNonNull with message Supplier argument

### PR DESCRIPTION
**Pull request by LeeTibbert**
Enable j.u.Objects#requireNonNull with message Supplier argument

Now that we have j.u.f.Supplier, the comment had become obsolete.
We can now implement that method.

### CODE REVIEW

File: javalib/src/main/scala/java/util/Objects.scala [hunk 1]
```diff
@@ -82,9 +84,8 @@ object Objects {
...
 
-  // Requires the implementation of java.util.function
-  // @inline
-  // def requireNonNull[T](obj: T, messageSupplier: Supplier[String]): T =
-  //   if (obj == null) throw new NullPointerException(messageSupplier.get())
-  //   else obj
+  @inline
+  def requireNonNull[T](obj: T, messageSupplier: Supplier[String]): T =
+     if (obj == null) throw new NullPointerException(messageSupplier.get())
+     else obj
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/Objects.scala:90]:
```suggestion
  def requireNonNull[T](obj: T, messageSupplier: Supplier[String]): T =
    if (obj == null) throw new NullPointerException(messageSupplier.get())
    else obj
```

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/ObjectsTest.scala [hunk 2]
```diff
@@ -94,6 +94,30 @@ class ObjectsTest {
...
+    val message =  "All is well!"
+
+    val successSupplier = new ju.function.Supplier[String] {
+          def get(): String = message
+      }
+
+    val failureSupplier = new ju.function.Supplier[String] {
+        def get(): String = throw new AssertionError(
+            "Objects.requireNonNull() should not have called Supplier")
+      }
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/ObjectsTest.scala:107]: A bunch of whitespace fixes:
```suggestion
    val message = "All is well!"

    val successSupplier = new ju.function.Supplier[String] {
      def get(): String = message
    }

    val failureSupplier = new ju.function.Supplier[String] {
      def get(): String = {
        throw new AssertionError(
            "Objects.requireNonNull() should not have called Supplier")
      }
    }
```

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/ObjectsTest.scala [hunk 3]
```diff
@@ -94,6 +94,30 @@ class ObjectsTest {
...
+
+    try {
+      ju.Objects.requireNonNull(null, successSupplier)
+      fail("Expected NullPointerException, got success")
+    } catch {
+      case e: NullPointerException => assertEquals(message, e.getMessage())
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/ObjectsTest.scala:113]: I believe you were looking for `AssertThrows.expectThrows`, to assert that an exception is thrown, and additionally get the instance to check its message in a subsequent `assertEquals`.

**Review by sjrd — CHANGES_REQUESTED** [review]
The commit message can be shortened to remove "personal" aspects from it. We tend to keep them factual and not relate to personal stories like motivation, history and goals. We can reference issue numbers or past commit shas. When we need a pronoun, we use "we". So perhaps here it could be:

    Enable j.u.Objects#requireNonNull with message Supplier argument

    Now that we have j.u.f.Supplier, the comment had become obsolete.
    We can now implement that method.

**Comment by LeeTibbert** [comment]
Changed as requested.  Thank you for the suggestions and concrete examples, especially for the commit message. AssertThrows.expectThrows was new to me and sure cuts down the code and makes it more readable.

**Review by sjrd — APPROVED** [review]

**Pull request merged**